src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -14,11 +14,24 @@
 
 namespace Microsoft.R.Host.Client.Session {
     public class RSessionProvider : IRSessionProvider, IRHostConnector {
+        private readonly IRSessionProviderCallback _callback;
         private readonly ConcurrentDictionary<Guid, RSession> _sessions = new ConcurrentDictionary<Guid, RSession>();
         private readonly DisposeToken _disposeToken = DisposeToken.Create<RSessionProvider>();
+        private readonly SemaphoreSlim _connectorSwitchLock = new SemaphoreSlim(1, 1);
+
+        private IRHostConnector _hostConnector;
         private int _sessionCounter;
 
-        private IRHostBrokerConnector BrokerConnector { get; } = new RHostBrokerConnector();
+        public bool IsRemote => _hostConnector.IsRemote;
+        public string Name => _hostConnector.Name;
+        public Uri BrokerUri => _hostConnector.BrokerUri;
+        public string BrokerName => _hostConnector.Name;
```

Comment:
This combo of Name/Uri/IsRemote starts appearing often, perhaps an interface would be handy